src/libhirte/common/network.c

```diff
@@ -0,0 +1,39 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+#include "network.h"
+
+bool isIPv4Addr(const char *variable) {
+        struct sockaddr_in sa;
+        return (inet_pton(AF_INET, variable, &(sa.sin_addr)) == 1);
+}
+
+bool isIPv6Addr(const char *variable) {
+        struct sockaddr_in6 sa;
+        return (inet_pton(AF_INET6, variable, &(sa.sin6_addr)) == 1);
+}
+
+int get_address(const char *domain, char *ip_address, size_t ip_address_size) {
+        struct addrinfo hints, *res = NULL;
+        memset(&hints, 0, sizeof(hints));
+        hints.ai_family = AF_UNSPEC;
+        hints.ai_socktype = SOCK_STREAM;
+
+        int status = getaddrinfo(domain, NULL, &hints, &res);
+        if (status != 0) {
+                return false;
+        }
+
+        void *addr = NULL;
+        if (res->ai_family == AF_INET) { // IPv4 address
+                struct sockaddr_in *ipv4 = (struct sockaddr_in *) res->ai_addr;
+                addr = &(ipv4->sin_addr);
+                inet_ntop(AF_INET, addr, ip_address, ip_address_size);
+        } else if (res->ai_family == AF_INET6) { // IPv6 address
+                struct sockaddr_in6 *ipv6 = (struct sockaddr_in6 *) res->ai_addr;
```

Comment:
Since we don't support IPv6 (yet), maybe we remove this part and add it only once we need/use it?

Reply:
I would keep it as, I intend to change that soon.